server/controllers/userController.js

```diff
@@ -1,107 +1,117 @@
+            return response.status(201).send({
               message: 'User was successfully created',
               token,
               data: user
             });
           })
           .catch((err) => {
-            res.status(400).send({
-              message: err.message
+            response.status(400).send({
+              message: 'There was a problem creating this user', err
             });
           });
       });
   },
 
-  loginUser(req, res) {
+  loginUser(request, response) {
     db.users
       .findOne({
         where: {
-          email: req.body.email
+          email: request.body.email
         }
       })
       .then((user) => {
-        if (user && user.validatePassword(req.body.password)) {
+        if (!user) {
+          return response.status(400).send({
+            message: 'User was not found'
+          });
+        }
+        if (!user.validatePassword(request.body.password)) {
+          return response.status(401).send({
+            message: 'Invalid password',
+          });
+        }
+        if (user && user.validatePassword(request.body.password)) {
           const payload = {
             userId: user.id,
             roleId: user.roleId
           };
           const token = jwt.sign(payload, secret, { expiresIn: expires });
-          return res.status(200).send({
-            message: 'You were successfully logged in',
+          response.status(200).send({
+            message: 'You are successfully logged in',
             token,
             expiresIn: expires
           });
+        } else {
+          response.status(401).send({
+            message: 'Invalid login credentials',
+          });
         }
       })
       .catch((err) => {
-        res.status(401).send({
-          message: `There was a problem while logging in ${err.message}`,
+        response.status(400).send({
+          message:
+          'There was a problem while logging in due to invalid credentials',
+          err
         });
       });
   },
 
-  findUserById(req, res) {
-    const userDetails = {
-      user: ['id', 'firstName', 'lastName', 'email', 'username'],
-      role: ['id', 'title']
-    };
-    const query = {
-      where: {
-        id: req.params.id
-      },
-      attributes: userDetails.user,
-      include: [
-        {
-          model: db.roles,
-          attributes: userDetails.role
-        }
-      ]
-    };
+  findUserById(request, response) {
     db.users
-      .findOne(query)
+      .findById(request.params.id)
       .then((user) => {
         if (user) {
-          return res.status(200).send({ message: 'User found!', data: user });
+          user.password = null;
+          return response.status(200).send({ message: 'User found!', data: user });
         }
       })
       .catch((err) => {
-        res.status(404).send({
-          message: `User ${req.params.id} was not found`
+        response.status(404).send({
+          message: 'User was not found', err
```

Comment:
Sequleize will always return successfully when a resource is not found. this message should be inside the then block after if(user), since the user will be null if not found.

Errors thrown when using find or findAll in sequelize can be caused by anything. 
Therefore, it will not be explicitly known by you, that just resolves to a 500 error.